components/script/dom/shadowroot.rs

```diff
@@ -326,9 +326,24 @@ impl VirtualMethods for ShadowRoot {
         let shadow_root = self.upcast::<Node>();
 
         shadow_root.set_flag(NodeFlags::IS_CONNECTED, context.tree_connected);
-        for node in shadow_root.children() {
+
+        for node in shadow_root.traverse_preorder(ShadowIncluding::No) {
             node.set_flag(NodeFlags::IS_CONNECTED, context.tree_connected);
-            node.bind_to_tree(context);
+
+            if context.tree_is_in_a_shadow_tree {
+                if let Some(shadow_root) = node.containing_shadow_root() {
+                    node.set_containing_shadow_root(Some(&*shadow_root));
+                }
+                debug_assert!(node.containing_shadow_root().is_some());
+            }
```

Comment:
I think this block does nothing?

Either way, all the shadowroots children should already be bound to the root, so I think you can remove this.